test/models/flava/test_flava.py

```diff
@@ -22,44 +22,71 @@
 NUM_CLASSES = 2
 
 
-class TestFLAVA(unittest.TestCase):
-    def setUp(self):
-        torch.manual_seed(1234)
+@pytest.fixture(autouse=True)
+def random():
+    set_rng_seed(1234)
 
-    @torch.no_grad()
-    def test_forward_classification(self):
+
+class TestFLAVA:
+    @pytest.fixture
+    def classification_inputs(self):
         text = torch.randint(0, 30500, (2, 77), dtype=torch.long)
         image = torch.rand((2, 3, 224, 224))
-
         labels = torch.randint(0, 2, (2,), dtype=torch.long)
+        return text, image, labels
+
+    @pytest.fixture
+    def pretraining_inputs(self):
+        text = torch.randint(0, 30500, (2, 77), dtype=torch.long)
+        image = torch.rand((2, 3, 224, 224))
+        image_for_codebook = torch.rand(2, 3, 112, 112)
+        image_patches_mask = torch.randint(0, 2, (2, 196), dtype=torch.long)
+        text_masked = text.detach().clone()
+        text_masked[:, 1:3] = 100
+        mlm_labels = text.detach().clone()
+        mlm_labels[:, :] = -1
+        mlm_labels[:, 1:3] = text[:, 1:3]
+        itm_labels = torch.tensor((0, 1), dtype=torch.long)
+        return (
+            text,
+            image,
+            image_for_codebook,
+            image_patches_mask,
+            text_masked,
+            mlm_labels,
+            itm_labels,
+        )
+
+    @torch.no_grad()
```

Comment:
In this case it may actually be useful to keep it; the pretraining test is using the full FLAVA model architecture and is one of our slower tests. I will remove the `no_grad` decorator and test locally as a sanity check though